packages/client-sdk-nodejs/src/internal/storage-data-client.ts

```diff
@@ -0,0 +1,337 @@
+  private readonly interceptors: Interceptor[];
+  private static readonly DEFAULT_MAX_SESSION_MEMORY_MB: number = 256;
+
+  /**
+   * @param {StorageClientPropsWithConfig} props
+   * @param dataClientID
+   */
+  constructor(props: StorageClientPropsWithConfig, dataClientID: string) {
+    this.configuration = props.configuration;
+    this.cacheServiceErrorMapper = new CacheServiceErrorMapper(
+      props.configuration.getThrowOnErrors()
+    );
+    this.credentialProvider = props.credentialProvider;
+    this.logger = this.configuration.getLoggerFactory().getLogger(this);
+    this.requestTimeoutMs = this.configuration
+      .getTransportStrategy()
+      .getGrpcConfig()
+      .getDeadlineMillis();
+    this.validateRequestTimeout(this.requestTimeoutMs);
+    this.logger.debug(
+      `Creating leaderboard client using endpoint: '${this.credentialProvider.getStorageEndpoint()}'`
+    );
+
+    // NOTE: This is hard-coded for now but we may want to expose it via StorageConfiguration in the
+    // future, as we do with some of the other clients.
+    const grpcConfig = new StaticGrpcConfiguration({
+      deadlineMillis: this.configuration
+        .getTransportStrategy()
+        .getGrpcConfig()
+        .getDeadlineMillis(),
+      maxSessionMemoryMb: StorageDataClient.DEFAULT_MAX_SESSION_MEMORY_MB,
+    });
+    const channelOptions = grpcChannelOptionsFromGrpcConfig(grpcConfig);
+
+    this.clientWrapper = new IdleGrpcClientWrapper({
+      clientFactoryFn: () =>
+        new store.StoreClient(
+          this.credentialProvider.getStorageEndpoint(),
+          this.credentialProvider.isStorageEndpointSecure()
+            ? ChannelCredentials.createSsl()
+            : ChannelCredentials.createInsecure(),
+          channelOptions
+        ),
+      loggerFactory: this.configuration.getLoggerFactory(),
+      maxIdleMillis: this.configuration
+        .getTransportStrategy()
+        .getMaxIdleMillis(),
+    });
+
+    const context: MiddlewareRequestHandlerContext = {};
+    context[CONNECTION_ID_KEY] = dataClientID;
+    this.interceptors = this.initializeInterceptors(
+      this.configuration.getLoggerFactory(),
+      this.configuration.getMiddlewares(),
+      context
+    );
+  }
+
+  close() {
+    this.logger.debug('Closing storage data clients');
+    this.clientWrapper.getClient().close();
+  }
+
+  private validateRequestTimeout(timeout?: number) {
+    this.logger.debug(`Request timeout ms: ${String(timeout)}`);
+    if (timeout !== undefined && timeout <= 0) {
+      throw new InvalidArgumentError(
+        'request timeout must be greater than zero.'
+      );
+    }
+  }
+
+  private initializeInterceptors(
+    _loggerFactory: MomentoLoggerFactory,
+    middlewares: Middleware[],
+    middlewareRequestContext: MiddlewareRequestHandlerContext
+  ): Interceptor[] {
+    const headers = [
+      new Header('Authorization', this.credentialProvider.getAuthToken()),
+      new Header('Agent', `nodejs:store:${version}`),
```

Comment:
is this header getting sent on every request

Reply:
No, the Header Interceptor class takes care of that:
https://github.com/momentohq/client-sdk-javascript/blob/main/packages/client-sdk-nodejs/src/internal/grpc/headers-interceptor.ts

Reply:
got it, thanks